### PR TITLE
feat: refactor `ftl init` and introduce `ftl new`

### LINKED ISSUE #1775: RFC: project initialization, module creation, and the Git Root
The current process to create an FTL project follows. The result is an `ftl-project.toml` file and a module `mymodule`.
```
git init .
ftl init go . mymodule
```

FTL currently makes an assumption that a project is in a Git repository, so commands such as `ftl init` will fail if `git init` is not run first. This made sense when the `ftl-project.toml` file wasn't required, since the root of the project was assumed to be the Git root, but when a single `ftl-project.toml` file is required, does this still make sense?

Up until recently, `ftl init` only created a module, since `ftl-project.toml` was not required. It now creates a `ftl-project.toml` file in addition to a module `mymodule`. Does it still make sense to overload the `init` command?

Proposal: separate out `ftl init` from a new command, something like `ftl module`.
```
ftl init .
ftl module go mymodule
```

Here, `ftl init` would create a `ftl-project.toml` file, and maybe create a git repository if one doesn't exist?
- Option 1: create a git repository, with `.gitignore` etc as we do now
- Option 2: prompt the user about creating a git repository
- Option 3: don't create a git repository, and no longer rely on there being a Git root
  - In this case, to find the project root, we'd walk up the dir tree until we find an `ftl-project.toml`

Does this seem cleaner?

### PR DESCRIPTION
**Pull request by safeer**
Fixes #1775

### CODE REVIEW

File: common/projectconfig/projectconfig.go [hunk 1]
```diff
@@ -99,24 +99,24 @@ func DefaultConfigPath() optional.Option[string] {
 	return optional.Some(filepath.Join(dir, "ftl-project.toml"))
 }
 
-// MaybeCreateDefault creates the ftl-project.toml file in the Git root if it
-// does not already exist.
-func MaybeCreateDefault(ctx context.Context) error {
+// CreateDefault creates the ftl-project.toml file with the given Config into dir.
+func CreateDefault(ctx context.Context, config Config, dir string) error {
```
Comment (alecthomas) [inline common/projectconfig/projectconfig.go:103]: This is not really the "default" anymore is it? Just rename it to `Create()`

**Review by alecthomas — APPROVED** [review]

File: cmd/ftl/cmd_init.go [hunk 2]
```diff
@@ -1,152 +1,69 @@
...
-	}
-	if token.Lookup(name).IsKeyword() {
-		return false
-	}
-	return true
+	Hermit       bool     `help:"Include Hermit language-specific toolchain binaries in the module." negatable:""`
+	Dir          string   `arg:"" help:"Directory to initialize the project in."`
+	ExternalDirs []string `help:"Directories of existing external modules."`
+	ModuleDirs   []string `help:"Child directories of existing modules."`
+	NoGit        bool     `help:"Do not initialize a git repository in the project directory."`
```
[the remark this answers is not in the document]
Comment (alecthomas) [reply cmd/ftl/cmd_init.go:23]: Related we should add a `no-git` field to the project config

[the remark this answers is not in the document]
Comment (safeer) [reply cmd/ftl/cmd_init.go:23]: Ahh yes, missed that point in our discussion in #1775

**Comment by safeer** [comment]
@alecthomas ptal

**Review by alecthomas — APPROVED** [review]

**Pull request merged**